src/main/java/lily/Ui.java

```diff
@@ -0,0 +1,241 @@
+                    + getCommands();
+        }
+        prettyPrint(welcomeMessage);
+    }
+
+    /**
+     * Shows an error within the styled boxes.
+     * 
+     * @param msg The error message to be shown.
+     */
+    public void showError(String msg) {
+        System.err.println(LS + INDENT + DIVIDER
+                + INDENT + msg + LS
+                + INDENT + DIVIDER);
+    }
+                            
+    /**
+     * Shows a message within the styled boxes.
+     * 
+     * @param s The message to be printed.
+     */
+    private void prettyPrint(String s) {
+        System.out.println(LS + INDENT + DIVIDER
+                + INDENT + s + LS
+                + INDENT + DIVIDER);
+    }
+
+    /**
+     * Lists the tasks in the form of a String for printing.
+     * 
+     * @return The tasks as Strings to be printed.
+     */
+    private String listTasks(LinkedList<Task> list) {
+        String listMsg = "";
+        int i = 1;
+        for (Task t : list) {
+            listMsg += "    " + i + "."
+                    + t.toString()
+                    + (i == list.size() ? "" : System.lineSeparator());
+            i++;
+        }
+        return listMsg;
+    }
+
+    /**
+     * Lists the actions users can take to interact with Lily.
+     * 
+     * @return A string with indents in front of actions users can take.
+     */
+    public String getCommands() {
+        return INDENT + "> todo: record a task which has no date" + LS
+                + INDENT + "> event: note an event with its date after /at" + LS
+                + INDENT + "> deadline: note something with its date after /by" + LS
+                + INDENT + "> list: show what you have to do" + LS
+                + INDENT + "> mark: indicate which numbered task you completed" + LS
+                + INDENT + "> unmark: indicate which task you havent completed" + LS
+                + INDENT + "> find: search your list for a single keyword" + LS
+                + INDENT + "> bye: stop talking with Lily";
+    }
+
+    /**
+     * Reads user input from the command line.
+     * 
+     * @return The user's input as a string.
+     */
+    public String readCommand() {
+        return userInput.nextLine();
+    }
+
+    /**
+     * Prints the exit message and closes the scanner.
+     */
+    public void closeUi() {
+        prettyPrint("finally. what took you so long? (´-ω-`)" + LS + LS
+                + INDENT + "oh, if your list had stuff, i've saved it" + LS
+                + INDENT + "i'll bring it up when you bug me again. bye");
+        userInput.close();
+    }
+
+    /**
+     * Displays the items in the TaskList to the user.
+     * 
+     * @param tl The TaskList to be shown.
+     */
+    public void showList(TaskList tl) {
+        if (tl.isEmpty()) {
+            showError("there's nothing in the list bro");
+        } else {
+            prettyPrint("you told me you had to" + LS + listTasks(tl.getContents()));
+        }
+    }
```

Comment:
Good use of camelCasing and descriptive verbs to name methods

Reply:
Thanks!